### DISCUSSION TITLE
Disable friendly-snippets when enabling blink.cmp?

### DISCUSSION THREAD

**Discussion by bbaldino** [LazyVim/LazyVim#5271] [Q&A] [answered]
I'm using blink.cmp but would like to disable friendly-snippets.  Is there a way to do this without having to copy the entire blink.cmp default config and disabling friendly-snippets there?  I try not to put any more custom stuff in the plugins dir than I have to.

**Comment by dpetka2001** [comment, marked as answer]
```lua
{
    "rafamadriz/friendly-snippets",
    enabled = false
  },
```
should do it I believe? Put it in one of your files in `/lua/plugins/` directory or in a new file with a `return` if you'd like.

**Reply by bbaldino** [reply]
Hm, gave that a shot but that doesn't appear to have done it

EDIT: I take that back...I reopened lazy and saw it did uninstall friendlysnippets, so perhaps the snippet I'm seeing is coming from somewhere else.

**Reply by dpetka2001** [reply]
LSP servers also send snippets.

**Reply by bbaldino** [reply]
Ah ok, maybe it's rustaceanvim sending it, I'll check if I can disable those. Or maybe can just disable all lap snippet sources

**Reply by dpetka2001** [reply]
You could do 
```lua
return {
	{
		"saghen/blink.cmp",
		opts = function(_, opts)
			opts.sources.default = { "lsp", "path", "buffer" }
		end,
	},
}
```
so that blink doesn't include the `snippets` source any more. Not entirely sure if that will also disable the lsp snippets, but you could try. Otherwise search for some setting in the LSP server to see if it provides such an option.

**Reply by bbaldino** [reply]
I do want to include my custom snippets (from luasnip) I just don't want them from anywhere else.

**Reply by dpetka2001** [reply]
Then you'll have to see if the LSP server provides such an option. Or maybe someone else has a better idea with how to deal with it.

**Reply by gennaro-tedesco** [reply]
You could also simply avoid installing friendly-snippet as dependency, if you don't need it :p

**Reply by bbaldino** [reply]
> You could also simply avoid installing friendly-snippet as dependency, if you don't need it :p

It's installed by default by other dependencies

**Comment by bbaldino** [comment]
I believe the snippets I was trying to disable come from rust analyzer's [magic completions](https://rust-analyzer.github.io/manual.html#magic-completions) and, in my relatively quick search, don't seem to be possible to disable.  But @dpetka2001 's instructions to disable friendly-snippets in lazyvim is correct :+1:

**Reply by abeldekat** [reply]
Hello @bbaldino,

Just my 2 cents...)

This is something I have been thinking about the last couple of days.
For me, the magic completions from rust-analyzer turn out to be very relevant, especially when adding ";" to a statement. See [this](https://github.com/echasnovski/mini.nvim/issues/1428#issuecomment-2564277980) comment in the beta testing issue of `mini.snippets`:

> Visualizing the tabstops is a great and distinctive feature! As a consequence, I discovered the power of having the lsp expand a function call with argument stops ("callSnippet" in "lua_ls"). I can use `<c-l>` during any function completion to navigate to the end of the function call. That behavior is similar to for example [tabout.nvim](https://github.com/abecodes/tabout.nvim%5D) and a real improvement when appending a ";" at the end of a function call in for example rust. No more `<esc>A;` or the like...

**Comment by dpetka2001** [comment]
@bbaldino It seems you can disable rust-analyzer's snippets as [per this comment](https://github.com/mrcjkb/rustaceanvim/discussions/545#discussioncomment-10962480). Try putting that in your `options.lua` file if you still want to go through with disabling snippets (along with disabling `friendly-snippets` as already suggested).
